### PR TITLE
Ref task 3364 Add singleton pattern for sign up screen

### PR DESCRIPTION
**Pull request by tuanda5353**
### Summary:
- [Feature] Private Room screen
### Link Issue:
- https://edu-redmine.framgia.vn/issues/3364

### CODE REVIEW

**Comment by tuanda5353** [comment]
ready

File: app/src/main/java/framgia/com/ichat/screen/signup/SingUpPresenter.java [hunk 1]
```diff
@@ -17,8 +17,17 @@
 public class SingUpPresenter implements SignUpContract.Presenter, OnCompleteListener, OnFailureListener {
     private SignUpContract.View mView;
     private AuthenticationRepository mRepository;
+    private static SingUpPresenter sInstance;
 
-    public SingUpPresenter(AuthenticationRepository repository) {
+    public static SingUpPresenter getInstance(AuthenticationRepository repository) {
+        if (sInstance == null) {
+            synchronized (SingUpPresenter.class) {
+                sInstance = new SingUpPresenter(repository);
```
Comment (DoanVanToan) [inline app/src/main/java/framgia/com/ichat/screen/signup/SingUpPresenter.java]: trong mvp thì chỉ 1 view liên lạc vs 1 presenter nên Presenter em k cần singleton

**Comment by DoanVanToan** [comment]
commented

**Comment by tuanda5353** [comment]
ready

**Review by DoanVanToan — APPROVED** [review]

**Pull request merged**